packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -190,6 +194,53 @@ export const useInvokeApi: EnsembleActionHook<InvokeAPIAction> = (action) => {
   return invokeApi;
 };
 
+const noneStyleOption = {
+  backgroundColor: "transparent",
+  showShadow: false,
+};
+
+export const useShowDialog: EnsembleActionHook<ShowDialogAction> = (
+  action?: ShowDialogAction,
+) => {
+  const { openModal } = useContext(ModalContext) || {};
+  const ensembleAction = useEnsembleAction(action?.onDialogDismiss);
+
+  const onDismissCallback = useCallback(() => {
+    if (!ensembleAction) {
+      return;
+    }
+    ensembleAction.callback();
+  }, [ensembleAction]);
+
+  if (!action?.widget)
+    throw new Error("ShowDialog Action requires a widget to be specified");
+
+  const callback = useCallback(() => {
+    const widget = unwrapWidget(action.widget);
```

Comment:
Unwrap once and memoize result. Does this work opening dialog multiple times? Might need to clone widget

Reply:
if by 'multiple times' you mean, opening dialog, closing it, then re-doing it again and again, then yes it works like that